wgpu-types/src/lib.rs

```diff
@@ -631,6 +631,19 @@ impl Limits {
             max_push_constant_size: 0,
         }
     }
+
+    /// Modify the current limits to use the 2D resolution limits of the other.
+    ///
+    /// This is useful because the swapchain might need to be larger than any other image in the application.
+    ///
+    /// If your application only needs 512x512, you might be running on a 4k display and need extremely high resolution limits.
+    pub fn using_resolution_2d(self, other: Self) -> Self {
```

Comment:
why don't we use the `default()` implementation limits here?

Reply:
Because if we use the default, rpi4 and such won't be able to run (as rpi has a limit of 4096x4096), but if we use 4k limits the guy with two 4k monitors will complain... so what we actually want is _just_ the adapter's limits.